02-express-tutorial/10-middleware-options

```diff
@@ -0,0 +1,34 @@
+const express = require('express')
+const app = express()
+const morgan = require('morgan')
+const logger = require('./logger')
+const authorize = require('./authorize')
+// req => middleware => res
+
+//app.use('/api',logger)//will invoke logger for any route, order matters, if no path argument is provided it's invoked for all paths
```

Comment:
the middleware logger will apply only when /api is reached. Try out  on "/"  endpoint, you will not get logger executed.
Think about like this:



const m2 = (req, res, next) =>{
  console.log("m2 method of the m2 middleware ")
  next();
}

const m3 = (req, res, next) =>{
  console.log("m3 middleware ")
  next();
}




app.use("/test", m3)
app.use("/test", m2)

In this situation, request to "/test" will call those two middleware functions.